ui/article-aem-renderer/src/main/java/org/cru/godtools/article/aem/service/AemArticleManager.kt

```diff
@@ -0,0 +1,316 @@
+
+    private val aemImportMutex = MutexMap()
+    private val articleMutex = MutexMap()
+    private val filesystemMutex = ReadWriteMutex()
+    private val resourceMutex = MutexMap()
+
+    // region Deeplinked Article
+    @AnyThread
+    suspend fun downloadDeeplinkedArticle(uri: Uri) {
+        aemDb.aemImportRepository().accessAemImport(AemImport(uri).apply { lastAccessed = Date() })
+        syncAemImport(uri, false)
+        downloadArticle(uri, false)
+    }
+    // endregion Deeplinked Article
+
+    // region Translations
+    @AnyThread
+    suspend fun syncAemImportsFromManifest(manifest: Manifest?, force: Boolean) = coroutineScope {
+        manifest?.aemImports?.forEach { launch { syncAemImport(it, force) } }
+    }
+
+    private val articleTranslationsJob = coroutineScope.launch {
+        val repository = aemDb.translationRepository()
+        dao.getAsFlow(QUERY_ARTICLE_TRANSLATIONS).conflate().collect { translations ->
+            translations.filterNot { repository.isProcessed(it) }.map { translation ->
+                launch {
+                    manifestManager.getManifest(translation)?.let { manifest ->
+                        repository.addAemImports(translation, manifest.aemImports)
+                        coroutineScope.launch { syncAemImportsFromManifest(manifest, false) }
+                    }
+                }
+            }.joinAll()
+
+            // prune any translations that we no longer have downloaded.
+            repository.removeMissingTranslations(translations)
+        }
+    }
+    // endregion Translations
+
+    // region AEM Import
+    /**
+     * This method is responsible for syncing an individual AEM Import url to the AEM Article database.
+     *
+     * @param baseUri The base AEM Import URL to sync
+     * @param force   This flag indicates that this sync should ignore the lastUpdated time
+     */
+    @WorkerThread
+    private suspend fun syncAemImport(baseUri: Uri, force: Boolean) {
+        if (!baseUri.isAbsolute || !baseUri.isHierarchical) return
+        aemImportMutex.withLock(baseUri) {
+            val aemImport = aemDb.aemImportDao().find(baseUri)?.takeIf { force || it.isStale() } ?: return
+
+            // fetch the raw json
+            val t = System.currentTimeMillis().let { if (force) it else it.roundTimestamp(CACHE_BUSTING_INTERVAL_JSON) }
+            val json = try {
+                api.getJson(baseUri.addExtension("9999.json"), t).takeIf { it.code() == HTTP_OK }?.body()
+            } catch (e: IOException) {
+                Timber.tag(TAG).v(e, "Error downloading AEM json")
+                null
+            } ?: return
+
+            // parse & store articles
+            val articles = json.findAemArticles(baseUri).toList()
+            aemDb.aemImportRepository().processAemImportSync(aemImport, articles)
+
+            // launch download of all the articles
+            articles.forEach { coroutineScope.launch { downloadArticle(it.uri, false) } }
```

Comment:
this statement is launching a separate coroutine to `downloadArticle()`, because it's using `launch` it's not expecting any return value here. So the answer to your questions would be contained within the logic of `downloadArticle()`